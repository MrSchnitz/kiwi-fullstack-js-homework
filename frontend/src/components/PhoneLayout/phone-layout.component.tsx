import React, { ReactNode } from "react";

interface PhoneLayoutProps {
  children: ReactNode;
}

const PhoneLayout: React.FC<PhoneLayoutProps> = ({
  children,
}: PhoneLayoutProps) => {
  return (
    <div className="device-google-pixel-2-xl">
      <div className="device-frame">
        <div className="device-content">{children}</div>
      </div>
      <div className="device-stripe" />
      <div className="device-header" />
      <div className="device-sensors" />
      <div className="device-btns" />
      <div className="device-power" />
    </div>
  );
};

export default PhoneLayout;
