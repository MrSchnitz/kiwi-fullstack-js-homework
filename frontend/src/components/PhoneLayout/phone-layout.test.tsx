import React from "react";
import {shallow, ShallowWrapper} from "enzyme";
import PhoneLayout, {PhoneLayoutProps} from "./phone-layout.component";

describe("PhoneLayout component", () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    const mockProps: PhoneLayoutProps = {
      children: <div>Component</div>,
    };

    wrapper = shallow(<PhoneLayout {...mockProps} />);
  });

  it("should render PhoneLayout component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
