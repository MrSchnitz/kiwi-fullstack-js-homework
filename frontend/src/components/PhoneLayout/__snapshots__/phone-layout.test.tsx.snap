// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneLayout component should render PhoneLayout component 1`] = `ShallowWrapper {}`;
